be/src/env/env_s3.cpp

```diff
@@ -77,15 +77,20 @@ StatusOr<std::unique_ptr<RandomAccessFile>> EnvS3::new_random_access_file(const
     std::string object = path.substr(namenode.size(), path.size() - namenode.size());
     Aws::Client::ClientConfiguration config;
     config.scheme = Aws::Http::Scheme::HTTP;
-    if (!config::aws_s3_endpoint.empty()) {
-        config.endpointOverride = config::aws_s3_endpoint;
+    if (is_oss_path(namenode.c_str())) {
+        assert(!config::object_storage_endpoint.empty());
+        config.endpointOverride = get_endpoint_from_oss_bucket(config::object_storage_endpoint, &bucket);
+    }
+    if (is_s3a_path(namenode.c_str()) && !config::object_storage_endpoint.empty()){
```

Comment:
Could it be better like? Not need to check `is_s3a_path`.

```
if (is_oss_path() {
...
} else if(!endpoint.empty()) {
...
}
```

Reply:
done